constraint/pkg/client/client.go

```diff
@@ -490,7 +525,7 @@ func (c *Client) AddConstraint(ctx context.Context, constraint *unstructured.Uns
 		resp.Handled[target] = true
 	}
 	if len(errMap) == 0 {
-		c.constraints[subPath] = constraint.DeepCopy()
+		c.constraints[constraint.GroupVersionKind().GroupKind()][subPath] = constraint.DeepCopy()
```

Comment:
should we check for empty GroupVersionKind for completeness

Reply:
getTemplateEntry() would have raised an error

Reply:
FWIW `getTemplateEntry()` only looks at the `Kind` so there's still the possibility of shenanigans.

Reply:
Technically an empty group is valid, though at that point it's not a Gatekeeper resource unless K8s changes.

I can add validation of the group to getTemplateEntry()

Reply:
added